spring-mvc-basics-4/src/main/java/com/baeldung/jsonparam/JsonParamController.java

```diff
@@ -0,0 +1,55 @@
+package com.baeldung.jsonparam;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+
+@Controller
+@RestController
+@RequestMapping("/api")
+public class JsonParamController {
+
+    @GetMapping(value = "/message")
+    public String testQueryParamApi(@RequestParam("jsonParam") String jsonParam) {
+        // jsonParam={"level":3,"type":"foo","color":"green"}
+        ParamObjectDTO paramObjectDTO;
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            paramObjectDTO = objectMapper.readValue(jsonParam, ParamObjectDTO.class);
+            System.out.println(paramObjectDTO);
+            // use paramObjectDTO where you have {"level":3,"type":"foo","color":"green"} JSON data as Object
+        } catch (JsonParseException e) {
+            e.printStackTrace();
```

Comment:
If we are going to catch the exceptions, let's use a Logger and also consider catching them all together on one line if the action is the same for all of them.